infrastructure/async/src/test/java/tech/pegasys/teku/infrastructure/async/ThrottlingTaskQueueTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.async;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.IntFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import tech.pegasys.teku.infrastructure.metrics.StubMetricsSystem;
+import tech.pegasys.teku.infrastructure.metrics.TekuMetricCategory;
+
+public class ThrottlingTaskQueueTest {
+
+  protected static final int MAXIMUM_CONCURRENT_TASKS = 3;
+
+  private final StubMetricsSystem stubMetricsSystem = new StubMetricsSystem();
+
+  private final StubAsyncRunner stubAsyncRunner = new StubAsyncRunner();
+
+  private final ThrottlingTaskQueue throttlingTaskQueue =
+      new ThrottlingTaskQueue(
+          MAXIMUM_CONCURRENT_TASKS, stubMetricsSystem, TekuMetricCategory.BEACON, "default");
+
+  private final ThrottlingTaskQueueWithPriority throttlingTaskQueueWithPriority =
```

Comment:
I'd have a dedicated ThrottlingTaskQueueWithPriorityTest test class, just to follow our standard